### ISSUE TITLE
Problems with timeouts after upgrading docker desktop to latest version

### ISSUE THREAD

**Issue by Sophietje** [TNO/knowledge-engine#311]
In GitLab by @dominic.ectors.vito.be on Oct 28, 2021, 10:05

I had to upgrade my docker desktop environment to the latest version, and after installing it I have a lot of issues with timeouts. No KB or KI registration was possible. The timeout value was 1000. 
I adapted the timeout environment parameter in the GA docker container, increased it to 10000 and that helped. But I still get timeouts now when waiting for an incoming ASK or POST request. As a result I get continuously 400 and 500 back as the response status.

Pinging from the GA container to the KER container showed no large round-trip delay.
```
--- 172.11.0.3 ping statistics ---
6 packets transmitted, 6 packets received, 0% packet loss
round-trip min/avg/max = 0.134/0.234/0.331 ms 
```

**Output log of GA container:**
```
09:47:11.782 [http-nio-9090-exec-9] INFO  e.i.a.controllers.SmartConnector - Registering / creating Smart Connector
09:47:12.024 [http-nio-9090-exec-1] INFO  e.i.a.controllers.SmartConnector - Registering Ask / Answer Knowledge Interaction
09:47:12.397 [http-nio-9090-exec-3] INFO  e.i.a.controllers.SmartConnector - Registering Post / React Knowledge Interaction
09:47:12.661 [http-nio-9090-exec-5] INFO  e.i.a.controllers.SmartConnector - Registering Post / React Knowledge Interaction
09:47:22.721 [http-nio-9090-exec-7] ERROR o.a.c.c.C.[.[.[.[dispatcherServlet] - Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is org.springframework.web.client.ResourceAccessException: I/O error on GET request for "http://172.11.0.3:8280/rest/sc/handle": Read timed out; nested exception is java.net.SocketTimeoutException: Read timed out] with root cause
java.net.SocketTimeoutException: Read timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedRead(Unknown Source)
	at java.base/sun.nio.ch.NioSocketImpl.implRead(Unknown Source)
	at java.base/sun.nio.ch.NioSocketImpl.read(Unknown Source)
	at java.base/sun.nio.ch.NioSocketImpl$1.read(Unknown Source)
	at java.base/java.net.Socket$SocketInputStream.read(Unknown Source)
	at org.apache.http.impl.io.SessionInputBufferImpl.streamRead(SessionInputBufferImpl.java:137)
	at org.apache.http.impl.io.SessionInputBufferImpl.fillBuffer(SessionInputBufferImpl.java:153)
	at org.apache.http.impl.io.SessionInputBufferImpl.readLine(SessionInputBufferImpl.java:280)
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:138)
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:56)
	at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:259)
	at org.apache.http.impl.DefaultBHttpClientConnection.receiveResponseHeader(DefaultBHttpClientConnection.java:163)
	at org.apache.http.impl.conn.CPoolProxy.receiveResponseHeader(CPoolProxy.java:157)
	at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:273)
	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:125)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:272)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)
	at org.springframework.http.client.HttpComponentsClientHttpRequest.executeInternal(HttpComponentsClientHttpRequest.java:87)
	at org.springframework.http.client.AbstractBufferingClientHttpRequest.executeInternal(AbstractBufferingClientHttpRequest.java:48)
	at org.springframework.http.client.AbstractClientHttpRequest.execute(AbstractClientHttpRequest.java:66)
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:775)
	at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:710)
	at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:601)
	at eu.interconnect.adapter.controllers.SmartConnector.startWaitingForHandleRequest(SmartConnector.java:495)
	at jdk.internal.reflect.GeneratedMethodAccessor68.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:197)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:141)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:106)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:894)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1060)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:962)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:898)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:626)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:733)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:542)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:143)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:374)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:888)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1597)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Unknown Source)
09:47:22.801 [http-nio-9090-exec-8] ERROR e.i.a.controllers.SmartConnector - An error occurs while waiting for ask or post
09:47:22.839 [http-nio-9090-exec-9] ERROR e.i.a.controllers.SmartConnector - An error occurs while waiting for ask or post
09:47:22.907 [http-nio-9090-exec-10] ERROR e.i.a.controllers.SmartConnector - An error occurs while waiting for ask or post
09:47:22.963 [http-nio-9090-exec-1] ERROR e.i.a.controllers.SmartConnector - An error occurs while waiting for ask or post
09:47:23.000 [http-nio-9090-exec-2] ERROR e.i.a.controllers.SmartConnector - An error occurs while waiting for ask or post
```

**Output log KER:**
```
} (issued id: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6/interaction/54117f60-7b69-48ed-ac1d-a65e4a18b291)


2021-10-28 09:47:12:660 +0000 [qtp1752203484-17] INFO KnowledgeInteractionLifeCycleApiServiceImpl - Knowledge interaction created in KB https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6: class ReactKnowledgeInteraction {
    class KnowledgeInteractionBase {
       knowledgeInteractionType: ReactKnowledgeInteraction
        communicativeAct: class CommunicativeAct {
            requiredPurposes: [https://www.tno.nl/energy/ontology/interconnect#InformPurpose]
            satisfiedPurposes: [https://www.tno.nl/energy/ontology/interconnect#InformPurpose]
}
        prefixes: null
    }
    argumentGraphPattern: ?asset <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <https://saref.etsi.org/core/device> 
    ?profile <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://ontology.tno.nl/interconnect/forecast/timeSeries> .
    ?profile <http://www.w3.org/2006/time#hasTime> ?profilePeriod .
    ?profilePeriod <http://www.w3.org/2006/time#hasBeginning> ?startTime .
    ?startTime <http://www.w3.org/2006/time#inXSDDateTimeStamp> ?startTS .
    ?profile <http://www.w3.org/2006/time#hasTemporalResolution> ?res .
    ?res <http://www.w3.org/2006/time#days> ?res_d .
    ?res <http://www.w3.org/2006/time#hours> ?res_h .
    ?res <http://www.w3.org/2006/time#minutes> ?res_m .
    ?res <http://www.w3.org/2006/time#seconds> ?res_s .
    ?profile <http://ontology.tno.nl/interconnect/forecast/hasSeriesOfDataPoint> ?dp .
    ?dp <http://ontology.tno.nl/interconnect/forecast/hasQuantity> ?q .
    ?q a <http://ontology.tno.nl/interconnect/ic_units/power> .
    ?q <http://www.ontology-of-units-of-measure.org/resource/om/hasValue> ?v .
    ?v <http://www.ontology-of-units-of-measure.org/resource/om/hasNumericalValue> ?nv .
    ?v <http://www.ontology-of-units-of-measure.org/resource/om/hasUnit> <http://ontology.tno.nl/interconnect/ic_units/kilowatt> .
    ?dp <http://ontology.tno.nl/interconnect/forecast/hasTimeStamp> ?dpts .
    resultGraphPattern: null
} (issued id: hytps://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6/interaction/c807ef98-7213-4997-8aac-5e1927071164)
2021-10-28 09:47:12:714 +0000 [qtp1752203484-14] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:22:793 +0000 [qtp1752203484-15] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:22:831 +0000 [qtp1752203484-19] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:22:906 +0000 [qtp1752203484-18] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:22:960 +0000 [qtp1752203484-16] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:22:993 +0000 [qtp1752203484-17] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:059 +0000 [qtp1752203484-14] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:099 +0000 [qtp1752203484-15] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:148 +0000 [qtp1752203484-19] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:180 +0000 [qtp1752203484-18] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:226 +0000 [qtp1752203484-16] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:269 +0000 [qtp1752203484-17] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:306 +0000 [qtp1752203484-14] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:359 +0000 [qtp1752203484-15] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:404 +0000 [qtp1752203484-19] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:455 +0000 [qtp1752203484-18] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:506 +0000 [qtp1752203484-16] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:538 +0000 [qtp1752203484-17] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:581 +0000 [qtp1752203484-14] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:612 +0000 [qtp1752203484-15] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:648 +0000 [qtp1752203484-19] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:680 +0000 [qtp1752203484-18] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:734 +0000 [qtp1752203484-16] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:787 +0000 [qtp1752203484-17] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:840 +0000 [qtp1752203484-14] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:869 +0000 [qtp1752203484-15] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:908 +0000 [qtp1752203484-19] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:23:957 +0000 [qtp1752203484-18] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:24:009 +0000 [qtp1752203484-16] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:24:055 +0000 [qtp1752203484-17] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:24:132 +0000 [qtp1752203484-14] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:24:203 +0000 [qtp1752203484-15] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:24:236 +0000 [qtp1752203484-19] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
2021-10-28 09:47:24:281 +0000 [qtp1752203484-18] INFO ReactiveApiServiceImpl - scHandleGet() called by KB: https://ke.interconnectproject.eu/adapter/ba95937f13a558c0d03e664b0d9ecf5d9c44e0d7b9c74cce9618957456c25ad6cb911b7d39118d33606c4908f1679094dd6f8eee535adc4b3ae8489595bd6de6
```

**Comment by Sophietje** [comment]
In GitLab by @barry.nouwt.tno.nl on Oct 28, 2021, 10:39

Might be related to: https://gitlab.inesctec.pt/interconnect/knowledge-engine/-/issues/290

**Comment by Sophietje** [comment]
In GitLab by @barry.nouwt.tno.nl on Oct 28, 2021, 11:01

Hi Dominic, thanks for reporting. Since you have access to the KE logs I assume you are hosting your own KE. Did you already use version 1.0.0 of the Knowledge Engine? Or could it be the case that the Docker Desktop upgrade triggered your docker to pull the `latest` Knowledge Engine image and the problem is actually related to the latest version of the Knowledge Engine (1.0.0). Regards, Barry

**Comment by Sophietje** [comment]
In GitLab by @dominic.ectors.vito.be on Oct 28, 2021, 11:21

Hi Barry,

It is indeed with the latest KE (1.0.0) but since it are just containers I also tried it with the older version and I had the same problem. Maybe it isn't a KE problem, but I do not know how to circumvent it or why it is happening.

**Comment by Sophietje** [comment]
In GitLab by @barry.nouwt.tno.nl on Oct 28, 2021, 11:40

Hi Dominic, it would be interesting to see if I can reproduce this behavior. Is the code available in some InterConnect repo? And what versions of Docker Desktop are you using?

**Comment by Sophietje** [comment]
In GitLab by @dominic.ectors.vito.be on Oct 28, 2021, 15:17

Hi Barry, 
thanks for looking into it. When connecting to the cloud KER, I actually had no problems. So I concluded it is not a KE problem. 
At a certain moment I noticed that timeout values were added as environment variables in the docker GA configuration. So they were set to 1000 (msec) by default in my docker environment. The GA to the cloud KER was still a previous version and did not have these environment parameters. That explained why connecting to the cloud version didn't give any problem.
So I set them to 10000 and most of my problems disappeared, except the long polling. And then I got it. I enlarged the timeout value to 60000. (larger then the 29 seconds time out for long polling), and now it is working.
